tests/erlang_tests/test_split_binary.erl

```diff
@@ -0,0 +1,59 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2024 Paul Guyot <pguyot@kallisys.net>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(test_split_binary).
+
+-export([
+    start/0,
+    id/1
+]).
+
+start() ->
+    ok = test_split_binary(),
+    0.
+
+test_split_binary() ->
+    {<<"a">>, <<"bcde">>} = split_binary(id(<<"abcde">>), 1),
+    {<<>>, <<"abcde">>} = split_binary(id(<<"abcde">>), 0),
+    {<<"abcde">>, <<>>} = split_binary(id(<<"abcde">>), 5),
+    ok =
+        try
+            _ = split_binary(id(<<"abcde">>), 6),
+            unexpected
+        catch
+            error:badarg -> ok
+        end,
+    ok =
+        try
+            _ = split_binary(id(<<"abcde">>), -1),
+            unexpected
+        catch
+            error:badarg -> ok
+        end,
+    ok =
+        try
+            _ = split_binary(id(<<>>), 1),
```

Comment:
What's the behavior for empty binary split at index 0?

Reply:
As I think of it now, calls for `id` are used to prevent optimizations – shouldn't we use external calls (`?MODULE:id`) to prevent function inlining? Didn't check if this particular optimization pass exists but Davide turned my attention to it in my other PR.

Reply:
In this case I wrote the test first so I confirm at least with OTP27 that the compiler isn't smart enough to inline out the call.